mijia/src/bluetooth/bleuuid.rs

```diff
@@ -0,0 +1,108 @@
+use uuid::Uuid;
+
+const BLUETOOTH_BASE_UUID: u128 = 0x00000000_0000_1000_8000_00805f9b34fb;
+const BLUETOOTH_BASE_MASK: u128 = 0x00000000_ffff_ffff_ffff_ffffffffffff;
+const BLUETOOTH_BASE_MASK_16: u128 = 0xffff0000_ffff_ffff_ffff_ffffffffffff;
+
+/// Convert a 32-bit BLE short UUID to a full 128-bit UUID by filling in the standard Bluetooth Base
+/// UUID.
+pub const fn uuid_from_u32(short: u32) -> Uuid {
+    Uuid::from_u128(BLUETOOTH_BASE_UUID | ((short as u128) << 96))
+}
+
+/// Convert a 16-bit BLE short UUID to a full 128-bit UUID by filling in the standard Bluetooth Base
+/// UUID.
+pub const fn uuid_from_u16(short: u16) -> Uuid {
```

Comment:
Can these both be added to the BleUuid trait and called like `Uuid::from_u16(0x25)`? This should work a bit like how you implement Default::default() on structs.

Reply:
Unfortunately it doesn't seem to be possible to have const functions on a trait. )-: